assets/app/scripts/controllers/builds.js

```diff
@@ -44,7 +44,43 @@ angular.module('openshiftConsole')
     watches.push(DataService.watch("buildConfigs", $scope, function(buildConfigs) {
       $scope.buildConfigs = buildConfigs.by("metadata.name");
       Logger.log("buildConfigs (subscribe)", $scope.buildConfigs);
-    }));    
+    }));
+
+    var associateRunningBuildToBuildConfig = function(buildsByBuildConfig) {
+      var buildConfigBuildsInProgress = {}; 
+      for (var buildConfigName in buildsByBuildConfig) {
+        var buildConfigBuilds = buildsByBuildConfig[buildConfigName];
+        buildConfigBuildsInProgress[buildConfigName] = {};
+        for (var buildName in buildConfigBuilds) {
+          var build = buildConfigBuilds[buildName];
+          var buildStatus = build.status;
+          if (buildStatus === "New" || buildStatus === "Pending" || buildStatus === "Running") {
+            buildConfigBuildsInProgress[buildConfigName][buildName] = build;
+          }
+        }
+      }
+      return buildConfigBuildsInProgress;
+    }
+
+    watches.push(DataService.watch("builds", $scope, function(builds, action, build){
```

Comment:
don't set this up as a separate watch, we are already watching builds, just update the callback on the existing watch